includes/class.llms.roles.php

```diff
@@ -387,22 +388,31 @@ public static function remove_roles() {
 		}
 
 		// Remove custom caps from the WP core admin role.
-		self::update_caps( $wp_roles->get_role( 'administrator' ), 'remove' );
+		self::update_caps( $wp_roles->get_role( 'administrator' ), 'remove', array( 'wp' ) );
 
 	}
 
 	/**
 	 * Update the capabilities for a given role
 	 *
 	 * @since 3.13.0
+	 * @since [version] Added `$exclude_group` parameter that allows excluding groups of caps from the update.
 	 *
-	 * @param WP_Role $role Role object.
-	 * @param string  $type Update type [add|remove].
+	 * @param WP_Role  $role           Role object.
+	 * @param string   $type           Update type [add|remove].
+	 * @param string[] $exclude_groups Array of groups to exclude.
 	 * @return void
 	 */
-	private static function update_caps( $role, $type = 'add' ) {
+	private static function update_caps( $role, $type = 'add', $exclude_groups = array() ) {
+
+		$role_caps = self::get_all_caps( $role->name );
+		if ( ! empty( $exclude_groups ) ) {
+			foreach ( $exclude_groups as $exclude_group ) {
+				unset( $role_caps[ $exclude_group ] );
+			}
+		}
```

Comment:
I think this can be written more succinctly (and by that I mean without having to run (essentially) the same loop twice by using a `continue` in the original loop.

```suggestion
```

Reply:
well actually it's not running the same loop twice... and the idea was that in most of that cases with my code we make only 1 check, and if true (basically once, only when deleting the plugin), we run a for over 1 element.
With your suggestion below, we make a check for each group, and in most of the cases that check is useless.
So I chose ugliness+less_executions over succinctness.

If we want succinctness what about:
```php
$role_caps = self::get_all_caps( $role->name );
$role_caps = empty( $exclude_groups ) ? $role_caps : array_intersect_key( $role_caps, array_flip( $exclude_groups ) );
```

what do you think?

Reply:
Sure that works too.

I just don't like the idea of a single function performing two similar loops when it doesn't have to.

Reply:
Ok I will probably go with the fancy `array_flip` one because it better feeds my ego.

Reply:
solved in 0779fe1